geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/tables/SpatioTemporalTable.scala

```diff
@@ -59,32 +71,40 @@ object SpatioTemporalTable extends Logging {
     val MIN_START = "\u0000"
     val MAX_END = "~"
 
-    val schema = getIndexSchema(sft)
-      .getOrElse(throw new Exception("Cannot delete ${sft.getTypeName}.  SFT does not have its index schema stored."))
+    val schema = getIndexSchema(sft).getOrElse {
+      val msg = s"Cannot delete ${sft.getTypeName}. SFT does not have its index schema stored."
+      throw new Exception(msg)
+    }
 
     val (rowf, _,_) = IndexSchema.parse(IndexSchema.formatter, schema).get
-    rowf.lf match {
-      case Seq(pf: PartitionTextFormatter, const: ConstantTextFormatter, r@_*) =>
-        // Build ranges using pf and const!
+    val planners = rowf.lf match {
+      case Seq(pf: PartitionTextFormatter, i: IndexOrDataTextFormatter, const: ConstantTextFormatter, r@_*) =>
+        // Build ranges using pf, ip and const!
         val rpp = RandomPartitionPlanner(pf.numPartitions)
+        val ip = IndexOrDataPlanner()
         val csp = ConstStringPlanner(const.constStr)
+        Seq(rpp, ip, csp)
 
-        val planner =  CompositePlanner(Seq(rpp, csp), "~")
-        val kp = planner.getKeyPlan(AcceptEverythingFilter, ExplainPrintln)
+      case _ =>
+        throw new RuntimeException(s"Cannot delete ${sft.getTypeName}. SFT has an invalid schema structure.")
+    }
+
+    val planner =  CompositePlanner(planners, "~")
+    val keyPlans =
+      Seq(true, false).map(indexOnly => planner.getKeyPlan(AcceptEverythingFilter, indexOnly, ExplainNull))
 
-        val rs: Seq[data.Range] = kp match {
-          case KeyRanges(ranges) =>
-            ranges.map { r =>
-              new org.apache.accumulo.core.data.Range(r.start + "~" + MIN_START, r.end + "~" + MAX_END)
-            }
-          case _ => logger.error(s"Keyplanner failed to build range properly."); Seq()
-        }
+    val ranges = keyPlans.map { kp =>
+      kp match {
```

Comment:
use flatMap and remove the match statement (just pass in an anonymous partial function) and it will be syntactically the same and cleaner